test/parallel/rolldown.withdraw.merkleRoot.test.ts

```diff
@@ -0,0 +1,472 @@
+  );
+
+  await signTx(
+    getApi(),
+    await Withdraw(
+      testUser,
+      BN_HUNDRED,
+      testUser.keyRingPair.address,
+      "Arbitrum",
+    ),
+    testUser.keyRingPair,
+  ).then((events) => {
+    const res = getEventResultFromMangataTx(events);
+    expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  try {
+    await Rolldown.waitForNextBatchCreated("Ethereum", 10);
+  } catch (err) {
+    error = err;
+  }
+  expect(error).toEqual(
+    "method rolldown.TxBatchCreated not found within blocks limit",
+  );
+});
+
+describe("Pre-operation withdrawal tests", () => {
+  beforeEach(async () => {
+    await Sudo.batchAsSudoFinalized(
+      ...(await getWithdrawalExtrinsics(batchSize)),
+    );
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    expect(event.source).toEqual("AutomaticSizeReached");
+  });
+
+  test("Given <batchSize> withdrawals WHEN they run successfully THEN a batch is generated AUTOMATICALLY from that L1, from ranges of (n,n+<batchSize>-1)", async () => {
+    const nextRequestId = JSON.parse(
+      JSON.stringify(await api.query.rolldown.l2OriginRequestId()),
+    );
+    await Sudo.batchAsSudoFinalized(
+      ...(await getWithdrawalExtrinsics(batchSize)),
+    );
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const sequencersList = await SequencerStaking.activeSequencers();
+    expect(sequencersList.toHuman().Ethereum).toContain(event.assignee);
+    expect(event.source).toEqual("AutomaticSizeReached");
+    expect(event.range.from.toNumber()).toEqual(nextRequestId.Ethereum);
+    expect(event.range.to.toNumber()).toEqual(
+      nextRequestId.Ethereum + (batchSize - 1),
+    );
+  });
+
+  test("Given a withdraw extrinsic run WHEN tokens are available THEN a withdraw will happen and l2Requests will be stored waiting for batch", async () => {
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, 10, gaspIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const l2Request = await Rolldown.getL2Request(event.range.to.toNumber());
+    expect(l2Request.withdrawal.tokenAddress).toEqual(gaspIdL1Asset.ethereum);
+    expect(l2Request.withdrawal.withdrawalRecipient).toEqual(
+      testUser.keyRingPair.address,
+    );
+  });
+
+  test("Given a withdraw extrinsic run WHEN token is Eth and user has some tokens THEN a withdraw will happen and update will be stored on l2Requests and will be stored waiting for batch", async () => {
+    await Sudo.batchAsSudoFinalized(
+      Assets.mintToken(ETH_ASSET_ID, testUser, BN_MILLION),
```

Comment:
try never reuse users, plz set them up in the test setup

Reply:
Done